AWSSDKSwiftCLI/Sources/SPR/UploadMetadata.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSS3
+import Smithy
+import SmithyStreams
+import AWSCLIUtils
+
+extension SPRPublisher {
+
+    func uploadMetadata() async throws {
```

Comment:
This method:
- Gets version metadata from S3 to verify it does not already exist, then
- Puts version metadata to S3.